### PR TITLE
Fixed #28051 -- Made migrations respect Index's name argument.

### PR DESCRIPTION
**Pull request by MarkusH**
Thanks Marc Tamlyn for the report.

### CODE REVIEW

File: docs/releases/1.11.1.txt [hunk 1]
```diff
@@ -9,4 +9,6 @@ Django 1.11.1 fixes several bugs in 1.11.
 Bugfixes
 ========
 
-* ...
+* Django 1.11 inadvertently set the name for an index during migrations even if
+  one was explicitly set. The fix will result rebuilding those indexes.
```
Comment (mjtamlyn) [inline docs/releases/1.11.1.txt]: Might be better as  `If you have created an explicitly named index with django 1.11, a new migration file will be created which will rebuild the index.`

Or at least `will result in rebuilding` is needed to read correctly

Comment (timgraham) [inline docs/releases/1.11.1.txt]: Suggested wording:
```
* Made migrations respect ``Index``’s ``name`` argument.  If you created a
  named index with Django 1.11, ``makemigrations`` will create a migration to
  recreate the index with the correct name (:ticket:`28051`).
```
(also suggested a similar commit message)

File: docs/releases/1.11.1.txt [hunk 2]
```diff
@@ -9,4 +9,6 @@ Django 1.11.1 fixes several bugs in 1.11.
 Bugfixes
 ========
 
-* ...
+* Made migrations respect ``Index``\'s ``name`` argument. If you created a
```
Comment (timgraham) [inline docs/releases/1.11.1.txt]: I don't think `\'` renders the quote in the correct direction, I used a right quote in my suggestion.

Comment (MarkusH) [inline docs/releases/1.11.1.txt]: Ah, sorry. I thought that was some fancy c&p magic that happened when posting to Github.

**Pull request merged**